rholang/src/test/scala/coop/rchain/rholang/externalservices/OllamaServiceSpec.scala

```diff
@@ -0,0 +1,142 @@
+package coop.rchain.rholang.externalservices
+
+import cats.effect.IO
+import cats.effect.concurrent.Ref
+import coop.rchain.metrics.{Metrics, NoopSpan, Span}
+import coop.rchain.shared.Log
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import coop.rchain.crypto.hash.Blake2b512Random
+import coop.rchain.models.Expr.ExprInstance.GString
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.models._
+import coop.rchain.rholang.externalservices.OllamaServiceMock
+import coop.rchain.rholang.interpreter.RhoRuntime.RhoISpace
+import coop.rchain.rholang.interpreter.InterpreterUtil._
+import coop.rchain.rholang.syntax._
+import coop.rchain.rspace.syntax._
+import org.scalatest._
+import org.scalatest.Matchers._
+
+import scala.collection.immutable.BitSet
+
+class OllamaServiceSpec extends FlatSpec with Matchers {
+
+  implicit val logF: Log[Task]            = new Log.NOPLog[Task]
+  implicit val noopMetrics: Metrics[Task] = new Metrics.MetricsNOP[Task]
+  implicit val noopSpan: Span[Task]       = NoopSpan[Task]()
+
+  val errorHandler                    = Ref.unsafe[IO, Vector[Throwable]](Vector.empty)
+  implicit val rand: Blake2b512Random = Blake2b512Random(Array.empty[Byte])
+
+  "Ollama chat process" should "work with model and prompt" in {
+    val contract =
+      """
+        |new ollama, return in {
+        |  ollama!("llama3.2", "What is 2+2?", *return)
+        |}
+      """.stripMargin
+
+    TestOllamaFixture.testOllama(
+      contract,
+      List(Expr(GString("Echo: What is 2+2?")))
+    )
+  }
+
+  it should "work with default model (prompt only)" in {
+    val contract =
+      """
+        |new ollama, return in {
+        |  ollama!("What is the meaning of life?", *return)
+        |}
+      """.stripMargin
+
+    TestOllamaFixture.testOllama(
+      contract,
+      List(Expr(GString("Echo: What is the meaning of life?")))
+    )
+  }
+
+  "Ollama generate process" should "work with model and prompt" in {
+    val contract =
+      """
+        |new generate, return in {
+        |  generate!("llama3.2", "Write a poem", *return)
+        |}
+      """.stripMargin
+
+    TestOllamaFixture.testOllamaGenerate(
+      contract,
+      List(Expr(GString("Generate: Write a poem")))
+    )
+  }
+
+  it should "work with default model (prompt only)" in {
+    val contract =
+      """
+        |new generate, return in {
```

Comment:
should it be like this?
```suggestion
        |new generate(`rho:ollama:generate`), return in {
```

Reply:
the same question about another cases